CRM/Core/Resources.php

```diff
@@ -809,4 +810,25 @@ public static function getEntityRefFilters() {
     return $filters;
   }
 
+  /**
+   * @param string $url
+   * @return string
+   */
+  public function addCacheCode($url) {
+    parse_str(parse_url($url, PHP_URL_QUERY), $queryParts);
```

Comment:
Thanks for doing the benchmark! 

`strpos` checking for both '?' and '&' covers all the cases I had in mind to check if the cache buster is already set so I'm happy to change it. 

However the `$existing` string is also used in the `str_replace` a bit further down (if it does exist) and that would require more `strpos` checks and a `substr` call. `$queryParts` is used below that too, which would require a second `substr` check at the least. I know those functions do seem a bit heavy handed, but they're doing a few things:

- finding if there is a query string part of the URL at all
- checking if either `?r=` or `&r=` is set
- finding the value of `r`

I'm sure it is possible with just using `strpos` and `substr` functions, just maybe not the cleanest or most performant.

If you think the performance hit is too much I can revert the `getUrl` method. One down side is splitting logic for query buster generation but the main focus of this PR is the custom CSS file so I'm totally open to reverting it.